### PR TITLE
Open source code for paper "Pointly-Supervised Instance Segmentation"

### PR DESCRIPTION
**Pull request by bowenc0221**
@ppwwyyxx 

Open-source Mask R-CNN with point-based supervision first.

Implicit PointRend will be in a separate PR since it could be further refined as discussed in https://github.com/facebookresearch/detectron2/issues/2975#issuecomment-827379225

### CODE REVIEW

File: projects/PointSup/train_net.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    If you'd like to do anything fancier than the standard training logic,
+    consider writing your own training loop (see plain_train_net.py) or
+    subclassing the trainer.
+    """
+    trainer = Trainer(cfg)
+    trainer.resume_or_load(resume=args.resume)
+    if cfg.TEST.AUG.ENABLED:
+        trainer.register_hooks(
+            [hooks.EvalHook(0, lambda: trainer.test_with_TTA(cfg, trainer.model))]
+        )
```
Comment (ppwwyyxx) [inline projects/PointSup/train_net.py:105]: remove this

File: projects/PointSup/tools/prepare_coco_point_annotations_without_masks.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+import numpy as np
+import os
+import sys
+import pycocotools.mask as mask_utils
+
+from detectron2.utils.env import seed_all_rng
+
+
+def get_point_annotations(input_filename, output_filename, num_points_per_instance):
+    with open(input_filename, "r") as f:
```
Comment (ppwwyyxx) [inline projects/PointSup/tools/prepare_coco_point_annotations_without_masks.py:16]: can you use `PathManager.open` in this file?

**Comment by bowenc0221** [comment]
Thanks for reviewing! I have addressed these two issues, PTAL.
